apps/api/src/services/data-gathering/data-gathering.service.ts

```diff
@@ -96,8 +97,9 @@ export class DataGatheringService {
         date
       );
 
-      const marketPrice =
-        historicalData[symbol][format(date, DATE_FORMAT)].marketPrice;
+      var marketPrice =
+        historicalData[symbol]?.[format(getYesterday(), DATE_FORMAT)]
```

Comment:
This breaks the original logic, no? 🤔

Let's introduce a `isDryRun` param that you know you just need to return the latest value (without upsert)

Reply:
```
if(isDryRun)
      {return historicalData} or {return marketPrice} ?
```

I need help in maintaining return type `MarketData`

Reply:
You can create a "synthetic" `MarketData` object, something like this:

```
if(isDryRun) {
  const date = new Date();
  
  ...

  return {
    dataSource,
    date,
    marketPrice,
    symbol,
    createdAt: null,
    id: null,
    state: 'INTRADAY'
  };
}
```

Reply:
This is working fine only if the `finally` block is removed. Somehow, even though the `isDryRun` is `true`, it was entering the `finally` block.